vanguard.go

```diff
@@ -127,14 +127,40 @@ func NewTranscoder(services []*Service, opts ...TranscoderOption) (*Transcoder,
 		unknownHandler: transcoderOpts.unknownHandler,
 		methods:        map[string]*methodConfig{},
 	}
+
+	var restOnlyServices []protoreflect.ServiceDescriptor
 	for _, svc := range services {
-		if err := transcoder.registerService(svc, defaultServiceOptions); err != nil {
+		resolvedOpts, err := transcoder.registerService(svc, defaultServiceOptions)
+		if err != nil {
 			return nil, err
 		}
+		if len(resolvedOpts.protocols) == 1 {
+			_, ok := resolvedOpts.protocols[ProtocolREST]
+			if ok {
+				restOnlyServices = append(restOnlyServices, svc.schema)
+			}
+		}
 	}
 	if err := transcoder.registerRules(transcoderOpts.rules); err != nil {
 		return nil, err
 	}
+
+	// Finally, check that any services with only REST as target protocol
+	// actually have at least one method with REST mappings.
+	for _, svcDesc := range restOnlyServices {
+		methods := svcDesc.Methods()
+		var numSupportedMethods int
+		for i, length := 0, methods.Len(); i < length; i++ {
+			methodDesc := methods.Get(i)
+			if transcoder.methods[methodPath(methodDesc)].httpRule != nil {
+				numSupportedMethods++
+			}
+		}
+		if numSupportedMethods == 0 {
+			return nil, fmt.Errorf("service %s only supports REST target protocol but has no methods with HTTP rules", svcDesc.FullName())
```

Comment:
I think we should instead support a default mapping of POST to `<service>/<method>`. See https://cloud.google.com/endpoints/docs/grpc/transcoding#where_to_configure_transcoding

Reply:
Do you just mean to use the Connect protocol in this case? If not, that's not a thing that actually works today (and smells like an enhancement beyond the scope of this PR).

Reply:
Not Connect but effectively. Think it should be valid to have zero annotations and point towards a REST service, so don't want the handler to error here (not to implement this new behaviour in this PR).

Reply:
> Think it should be valid to have zero annotations and point towards a REST service, so don't want the handler to error here

But it's _not_ actually valid yet. As a service to users, I think we should do as much as possible up-front to fail fast and give good error messages. That removes footguns from users and gives them clear information about why things aren't working.

You can always remove this check later, Once this repo actually supports what you're talking about. Though some sort of check seems useful to keep even then, like in case the service only has streaming endpoints (that don't use `HttpBody`).